store/siren-action-behavior.html

```diff
@@ -122,17 +136,17 @@
 			} else {
 				fields = this.getSirenFields(action);
 			}
-			if (action.method !== 'GET' && action.method !== 'HEAD') {
-				body = fields;
-			}
 
-			if (body && action.type.indexOf('json') !== -1) {
+			if (action.type.indexOf('json') !== -1) {
 				var json = {};
-				body.keys().forEach(function(key) {
-					json[key] = action.getFieldByName(key).value;
-				});
+				for (var i = 0; i < fields.length; i++) {
+					var field = fields[i];
+					json[field.name] = field.value;
+				}
 				headers.set('Content-Type', action.type);
 				body = JSON.stringify(json);
+			} else if (action.method !== 'GET' && action.method !== 'HEAD') {
+				body = new FormData(fields);
```

Comment:
I don't think you can initialize a `FormData` like this. It needs a `<form>` element or you need to iterate throw the fields and `append` them to the formData object

Reply:
thanks